pkg/querier/tripperware/build_info.go

```diff
@@ -0,0 +1,49 @@
+package tripperware
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/prometheus/common/version"
+	v1 "github.com/prometheus/prometheus/web/api/v1"
+)
+
+type buildInfoResponse struct {
+	Status string                `json:"status"`
+	Data   *v1.PrometheusVersion `json:"data"`
+}
+
+// BuildInfoRoundTripper is a RoundTripper that returns static build info information.
+type BuildInfoRoundTripper struct{}
+
+// NewBuildInfoRoundTripper creates a round tripper which returns build info.
+func NewBuildInfoRoundTripper() *BuildInfoRoundTripper {
+	return &BuildInfoRoundTripper{}
+}
+
+// RoundTrip implements RoundTripper.
+func (r *BuildInfoRoundTripper) RoundTrip(_ *http.Request) (*http.Response, error) {
```

Comment:
Instead of creating this `RoundTripper` couldnt't we just register another handler here:

https://github.com/cortexproject/cortex/blob/fa35f6f6af301577a1f5bc0e2f8ef8d81ac06fdc/pkg/api/api.go#L377

Reply:
Updated. PTAL